sort/dutchsationalslagsort.go

```diff
@@ -0,0 +1,25 @@
+// Implementation of Dutch National Flag Sort algorithm
+// Reference: https://en.wikipedia.org/wiki/Dutch_national_flag_problem
+
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+// DutchNationalFlagSort sorts an array of integers with the Dutch National Flag algorithm
+func DutchNationalFlagSort[T constraints.Signed](arr []T) []T {
```

Comment:
This algorithm seems more a category classification algorithm than a sorting algorithm (even though it looks very similar to quick sort, I think). It might be worth it moving to the `other` package instead of adding it into the `sort` package instead. What do you think @raklaptudirm @siriak ??

Reply:
I agree, it can only sort arrays with 3 distinct elements, so I wouldn't call it a sorting algorithm